File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/graphics/IGraphicsObject.java
package com.agrogames.islandsofwar.engine.abs.graphics;

import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject;
import com.agrogames.islandsofwar.engine.abs.updatable.UpdatableObject;

public interface IGraphicsObject extends RenderableObject, UpdatableObject {
    boolean shouldBeRemoved();
    String getSpawnSound();
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/graphics/IGraphicsObject.kt
package com.agrogames.islandsofwar.engine.abs.graphics

import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject
import com.agrogames.islandsofwar.engine.abs.updatable.UpdatableObject

interface IGraphicsObject : RenderableObject, UpdatableObject {
    fun shouldBeRemoved(): Boolean
    val spawnSound: String?
}